doc/mgmt/OpenConfig_Platform.md

```diff
@@ -0,0 +1,454 @@
+           |  +--ro oc-transceiver:vendor-part?         string
+           |  +--ro oc-transceiver:serial-no?           string
+           |  +--ro oc-transceiver:date-code?           oc-yang:date-and-time
+           |  +--ro oc-transceiver:supply-voltage
+           |  |  +--ro oc-transceiver:instant?          decimal64
+           |  +--ro oc-transceiver:laser-bias-current
+           |     +--ro oc-transceiver:instant?          decimal64
+           +--rw oc-transceiver:physical-channels
+              +--rw oc-transceiver:channel* [index]
+                 +--ro oc-transceiver:state
+                    +--ro oc-transceiver:laser-temperature
+                    |  +--ro oc-transceiver:instant?    decimal64
+                    +--ro oc-transceiver:output-power
+                    |  +--ro oc-transceiver:avg?        decimal64
+                    +--ro oc-transceiver:input-power
+                    |  +--ro oc-transceiver:instant?    decimal64
+                    +--ro oc-transceiver:laser-bias-current
+                       +--ro oc-transceiver:instant?    decimal64
+```  
+
+# Definition/Abbreviation
+### Table 1: Abbreviations
+| **Term**                 | **Definition**                         |
+|--------------------------|-------------------------------------|
+| YANG                     | Yet Another Next Generation: modular language representing data structures in an XML tree format        |
+| gNMI                     | gRPC Network Management Interface: used to retrieve or manipulate the state of a device via telemetry or configuration data         |
+| XML                     | eXtensible Markup Language   |
+
+# 1 Feature Overview
+## 1.1 Requirements
+### 1.1.1 Functional Requirements
+1. Provide support for OpenConfig YANG models.
+2. Support telemetry for following read only parameters on transceiver components via REST and gNMI.
+3. Add support for following features:
+- serial-no
+- connector-type
+- vendor
+- vendor-part
+- date-code
+- supply-voltage
+- laser-temperature
+- output-power
+- input-power
+- laser-bias-current
+
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+SONiC already supports telemetry for parameters like packet counters with methods such as Get via REST and gNMI. This feature adds support for telemetry for states of transceiver components with gNMI and YANG models of openconfig-transceiver.yang and with REST based on translib infra.
+
+### 1.2.2 Container
+The code changes for this feature are part of *Management Framework* container which includes the REST server and *gnmi* container for gNMI support in *sonic-mgmt-common* repository.
+
+### 1.2.3 The name of components
+openconfig-platform.yang does not hierarchically define their components.
+This design uses the name comprised of a prefix of " transceiver_" and the interface name managed by SONiC to specify the path of the target transceiver components for REST/gNMI without the the name conflicts(e.g., “transceiver_Ethernet0”).
```

Comment:
Discuss what happens when a breakout port is specified. It's a logical port, not necessarily physical, so what happens? (e.g., if I have a 100G port broken out into 4x25G and they're called Ethernet0-3, displaying transceiver_Ethernet2 will display what?)
- Display channel value(s) for that logical port only?
- Display channel values for all physical channels for the transceiver that includes that logical port? 

Consider breakout modes that cover multiple channels as well (e.g., 2x50G, 4x100G, etc.).

Reply:
Given that the component is a physical transceiver it seems intuitive that there would be one name for each module.  If multiple names map to the same module (e.g. four logical interfaces in a 4x100g breakout) it may be difficult to perform operations like "query all transceivers" since duplicates would be returned.
Also note that transceivers are field replaceable units and often do need to be replaced.  There is value in having a naming scheme which can be directly tied to the labeling of cages/connectors on the front of a switch which is usually consecutive numbering (e.g. 1..32 or 0..63).  The fewer translations and re-naming steps required to go between telemetry for a transceiver and finding that transceiver on a switch, the fewer opportunities for mistakes there will be.

Reply:
Ok, so it would be:
- Return nothing, since we don't support non-physical ports. 

That's fine with me, so please clarify that behavior in this document.